protoc-gen-twirp/generator.go

```diff
@@ -538,7 +538,7 @@ func (t *twirp) generateUtils() {
 	t.P(`  errorCode := `, t.pkgs["twirp"], `.ErrorCode(tj.Code)`)
 	t.P(`  if !`, t.pkgs["twirp"], `.IsValidErrorCode(errorCode) {`)
 	t.P(`    msg := "invalid type returned from server error response: "+tj.Code`)
-	t.P(`    return `, t.pkgs["twirp"], `.InternalError(msg)`)
+	t.P(`    return `, t.pkgs["twirp"], `.InternalError(msg).WithMeta("body", string(respBodyBytes))`)
```

Comment:
The error handling on this line doesn't distinguish between these two situations:
1.) The response body matches the shape of `twerrJSON`, but it has invalid data (such as an invalid error code)
2.) The response body does not match the shape of `twerrJSON` (maybe it's some other JSON response)

In the issue accompanying this PR, it seemed like 2) was relevant to your use case, but it seems like 1) was the original intent of this line. It may be a good idea to distinguish between those two cases, such that 2.) can reuse `twirpErrorFromIntermediary`. That helper function seems appropriate for that situation, and it includes additional metadata that would be useful for debugging.

Am I thinking about this correctly, @marwan-at-work?

Reply:
@dpolansky actually my case is the one in `1` you mentioned above :) 
For case 2, we already do that because the json decoder does `DisallowUnknownFields` and also checks for `Code == ''` and `twirpErrorFromIntermediary` is indeed called. 

Would you still that function to be called? In other words: 

```
	errorCode := twirp.ErrorCode(tj.Code)
	if !twirp.IsValidErrorCode(errorCode) {
		msg := "invalid type returned from server error response: " + tj.Code
		return twirpErrorFromIntermediary(statusCode, msg, string(respBodyBytes))
	}
```

Let me know and I'm happy to do it :v:

Reply:
Thanks for clarifying! I think this is fine as is for now.